### ISSUE TITLE
IndexError in download_from_nhl_stats_api.py

### ISSUE THREAD

**Issue by capitalg** [shane-miller/Fantasy-NHML#3]
Error message:

```
➜  Fantasy-NHML git:(master) python3 Data/DataDownloadScript/download_from_nhl_stats_api.py --g=6 --a=4 --ppp=2 --sog=1 --w=6 --ga=-1 --sv=0.25 --so=5
Processing Centers:  40%|██▊    | 4/10 [05:34<08:22, 83.74s/it]
Traceback (most recent call last):
  File "Data/DataDownloadScript/download_from_nhl_stats_api.py", line 449, in <module>
    main()
  File "Data/DataDownloadScript/download_from_nhl_stats_api.py", line 386, in main
    center_records = api_main(base_skater_url, center_tag, skater_report_list, num_years)
  File "Data/DataDownloadScript/download_from_nhl_stats_api.py", line 138, in api_main
    temp = api_helper(base_url, tag, report_list, season_id)
  File "Data/DataDownloadScript/download_from_nhl_stats_api.py", line 171, in api_helper
    records.get('data')[k].update(temp.get('data')[k])
IndexError: list index out of range
```

**Comment by shane-miller** [comment]
I can't seem to replicate this error. Does the same error occur if you try to run the code again? If it does work running it a second time, it's possible that the NHL's API didn't return all the data it was supposed to. I've had issues with their API failing to provide the correct data every now and then, though it does only happen rarely.

**Comment by capitalg** [comment]
I didn't have time to reproduce the issue, but I changed line 168 in `download_from_nhl_stats_api.py` to the following and it worked:

`if records.get('data') is None or len(records.get('data')) < total_length:`

I also had to remove all the code related to removing `__pycache__` folders; it was causing `run_all.py` to fail because `__pycache__` folders did not exist. 

I will submit a pull request if I get some time later this week. Overall, after these changes, the app worked like a charm and helped me greatly with my draft yesterday. Great work! :)

**Comment by shane-miller** [comment]
I'm glad that you found it useful.

I don't think that change you made is for the better though. After reviewing it, it seems it would potentially hurt the integrity of the data. The way the code works for the query is it collects data from various report types and then concats them all together for each player. The implication being made by the error that was thrown is that the NHL API returned a different number of players in a given season for one of the reports. I've had this exact problem with the NHL API before, but it rarely happens and I assume you got unlucky when you ran it the first time. After you added the line, I suspect that the API ended up returning the correct thing that time and never actually utilized the line you added. If the API returns what it's supposed to, the length would either be none or exactly equivalent to total_length. In the case that neither of those are correct, then it would need to query the API for the same thing again to get the correct info. This is my fault for not implementing that error correction though; I hadn't thought of that case when originally designing this.

**Comment by shane-miller** [comment]
I believe this issue should now be fixed with commit 258d017c4f47100d6a56e05e55a165f41f1bed04. In this commit I implemented the error correction I described in my previous comment.
